core/node/libp2p/routing.go

```diff
@@ -179,3 +184,80 @@ func PubsubRouter(mctx helpers.MetricsCtx, lc fx.Lifecycle, in p2pPSRoutingIn) (
 		},
 	}, psRouter, nil
 }
+
+func AutoRelayFeeder(cfgPeering config.Peering) func(fx.Lifecycle, host.Host, chan peer.AddrInfo, *ddht.DHT) {
+	return func(lc fx.Lifecycle, h host.Host, peerChan chan peer.AddrInfo, dht *ddht.DHT) {
+		ctx, cancel := context.WithCancel(context.Background())
+		done := make(chan struct{})
+
+		defer func() {
+			if r := recover(); r != nil {
+				fmt.Println("Recovering from unexpected error in AutoRelayFeeder:", r)
+				debug.PrintStack()
+			}
+		}()
+		go func() {
+			defer close(done)
+
+			// Feed peers more often right after the bootstrap, then backoff
+			bo := backoff.NewExponentialBackOff()
+			bo.InitialInterval = 15 * time.Second
+			bo.Multiplier = 3
+			bo.MaxInterval = 1 * time.Hour
+			bo.MaxElapsedTime = 0 // never stop
+			t := backoff.NewTicker(bo)
+			defer t.Stop()
+			for {
+				select {
+				case <-t.C:
+				case <-ctx.Done():
+					return
+				}
+
+				// Always feed trusted IDs (Peering.Peers in the config)
+				for _, trustedPeer := range cfgPeering.Peers {
+					if len(trustedPeer.Addrs) == 0 {
+						continue
+					}
+					select {
+					case peerChan <- trustedPeer:
```

Comment:
:information_source: this allows people to reuse already trusted peers from `Peering.Peers` in addition to peers discovered via `dht.WAN.GetClosestPeers`